### PR TITLE
Add documentation for the package command

### PR DESCRIPTION
**Pull request by jamesls**
This was the last action item to close out https://github.com/awslabs/chalice/issues/183

cc @kyleknap @stealthycoin

### CODE REVIEW

**Comment by jamesls** [comment]
>  Do you think it would be worthwhile to add a small helper script to chalice to make SAM deployments a bit less tedious?

Personally no.  This is more of an advanced feature if you have existing cfn tooling you prefer to use *or* if you want to integrate with codepipeline (in which case the generate-pipeline command has helper stuff for you).

If we were going to go down that route, I'd prefer we spend the time to just have a cfn based backend, but it's not something I think we should do right at this moment.

**Comment by stealthycoin** [comment]
You mention pipeline support but don't link to or elaborate on it, I think if it is mentioned in the header of the section it should be elaborated upon at some point or linked to another section at least.

**Comment by jamesls** [comment]
@stealthycoin Yeah that's because there is no pipeline topic guide yet.  I think that's something we should add but out of scope for this PR.

File: docs/source/topics/cfn.rst [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    $ . /tmp/venv/bin/activate
+    $ pip install chalice awscli
+    $ chalice new-project test-cfn-deploy
+    $ cd test-cfn-deploy
+
+At this point we've installed chalice and the AWS CLI and we have
+a basic app created locally.  Next we'll run the ``package`` command
+and look at its contents::
+
+    $ $ chalice package /tmp/packaged-app/
```
Comment (pickfire) [inline docs/source/topics/cfn.rst]: $ $

**Pull request merged**